pallets/relayer/src/lib.rs

```diff
@@ -0,0 +1,303 @@
+        Self::unsafe_add_auth_for_paying_key(paying_did, user_key, paying_key, 0u128.into());
+        Ok(())
+    }
+
+    fn base_accept_paying_key(origin: T::Origin, auth_id: u64) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender: user_key, ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+        let signer = Signatory::Account(user_key);
+
+        <Identity<T>>::accept_auth_with(&signer, auth_id, |data, auth_by| -> DispatchResult {
+            let (user_key, paying_key, polyx_limit) = match data {
+                AuthorizationData::AddRelayerPayingKey(user_key, paying_key, polyx_limit) => {
+                    Ok((user_key, paying_key, polyx_limit))
+                }
+                _ => Err(AuthorizationError::BadAuthType),
+            }?;
+
+            Self::auth_accept_paying_key(signer.clone(), auth_by, user_key, paying_key, polyx_limit)
+        })
+    }
+
+    fn base_remove_paying_key(
+        origin: T::Origin,
+        user_key: T::AccountId,
+        paying_key: T::AccountId,
+    ) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender,
+            primary_did: sender_did,
+            ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+
+        // allow: `origin == user_key` or `origin == paying_key`
+        if sender != user_key && sender != paying_key {
+            // allow: `origin == primary key of user_key's identity`
+            ensure!(
+                <Identity<T>>::get_identity(&user_key) == Some(sender_did),
+                Error::<T>::NotAuthorizedForUserKey
+            );
+        }
+
+        // Check if there is a paying key.
+        ensure!(
+            <Subsidies<T>>::contains_key(&user_key),
+            Error::<T>::NoPayingKey
+        );
+
+        // Get current Subsidy
+        let subsidy = <Subsidies<T>>::get(&user_key);
+
+        // Check if the current paying key matches.
+        ensure!(subsidy.paying_key == paying_key, Error::<T>::NotPayingKey);
+
+        // Remove paying key for user key.
+        <Subsidies<T>>::remove(&user_key);
+
+        Ok(())
+    }
+
+    fn base_update_polyx_limit(
+        origin: T::Origin,
+        user_key: T::AccountId,
+        polyx_limit: T::Balance,
+    ) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender: paying_key, ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+
+        // Check if there is a paying key.
+        ensure!(
+            <Subsidies<T>>::contains_key(&user_key),
+            Error::<T>::NoPayingKey
+        );
+
+        // Get current Subsidy
+        let subsidy = <Subsidies<T>>::get(&user_key);
+
+        // Check if the current paying key matches.
+        ensure!(subsidy.paying_key == paying_key, Error::<T>::NotPayingKey);
+
+        // Update polyx limit
+        <Subsidies<T>>::mutate(&user_key, |subsidy| {
+            (*subsidy).remaining = polyx_limit;
```

Comment:
```suggestion
            subsidy.remaining = polyx_limit;
```
The field projection "operator" `.` handles derefs: https://github.com/rust-lang/rust/blob/b6f3cb9502b1910f6af32f426fdb78e813b390ef/compiler/rustc_typeck/src/check/expr.rs#L1553-L1580